rma/models/rma.py

```diff
@@ -0,0 +1,772 @@
+    warranty_state = fields.Selection([('valid', _("Valid")),
+                                       ('expired', _("Expired")),
+                                       ('undefined', _("Undefined"))],
+                                      string='Warranty',
+                                      compute=_compute_warranty)
+    is_dropship = fields.Boolean(related="operation_id.is_dropship")
+    related_rma_line = fields.Many2one(
+        'rma.order.line', string='Parent RMA line', ondelete='cascade')
+    # rma_ids = fields.One2many('rma.order.line', 'parent_rma')
+    partner_address_id = fields.Many2one(
+        'res.partner', readonly=True,
+        states={'draft': [('readonly', False)]},
+        string='Partner Address',
+        help="This address of the supplier in case of Customer RMA operation "
+             "dropship. The address of the customer in case of Supplier RMA "
+             "operation dropship")
+    product_qty = fields.Float(
+        string='Ordered Qty', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True,
+        states={'draft': [('readonly', False)]})
+    qty_to_receive = fields.Float(
+        string='Qty To Receive',
+        digits=dp.get_precision('Product Unit of Measure'),
+        compute=_compute_qty_to_receive, store=True)
+    qty_incoming = fields.Float(
+        string='Incoming Qty', copy=False,
+        readonly=True, digits=dp.get_precision('Product Unit of Measure'),
+        compute=_compute_qty_incoming, store=True)
+    qty_received = fields.Float(
+        string='Qty Received', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        compute=_compute_qty_received,
+        store=True)
+    qty_to_deliver = fields.Float(
+        string='Qty To Deliver', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute=_compute_qty_to_deliver,
+        store=True)
+    qty_outgoing = fields.Float(
+        string='Outgoing Qty', copy=False,
+        readonly=True, digits=dp.get_precision('Product Unit of Measure'),
+        compute=_compute_qty_outgoing,
+        store=True)
+    qty_delivered = fields.Float(
+        string='Qty Delivered', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute=_compute_qty_delivered,
+        store=True)
+    qty_to_refund = fields.Float(
+        string='Qty To Refund', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'), readonly=True,
+        compute=_compute_qty_to_refund, store=True)
+    qty_refunded = fields.Float(
+        string='Qty Refunded', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute=_compute_qty_refunded, store=True)
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        if not self.invoice_id:
+            return
```

Comment:
Hi @aheficent , is this by design? If RMA unrelated to invoice then onchange will not fill **name** field. Meanwhile field **name** is readonly thus data can not be saved.